public/bitemp_graph.js

```diff
@@ -695,6 +730,59 @@ var barChart = function() {
       //top horizontal line
       lineCreator(0, width - margin.left, 3, 3, dragDown, 'dragDown');
       $('#endValBox').val(format(yScale.invert(0)));
+      
+      writeQuery();
+      createFilledRectangle();
+
+      function createFilledRectangle() {
+        $('#filledRect').remove();
+          x1 = $('#dragLeft')[0].__data__.x;
+          x2 = $('#dragRight')[0].__data__.x;
+          y1 = $('#dragUp')[0].__data__.y;
+          y2 = $('#dragDown')[0].__data__.y;
```

Comment:
These names could be more descriptive, or maybe a comment explaining what these things are